airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -63,19 +75,38 @@ def create_from_stream(cls, stream: Stream, source: AbstractSource, logger: logg
         message_repository = source.message_repository
         return StreamFacade(
             ThreadBasedConcurrentStream(
-                partition_generator=StreamPartitionGenerator(stream, message_repository),
+                partition_generator=StreamPartitionGenerator(
+                    stream,
+                    message_repository,
+                    SyncMode.full_refresh if isinstance(cursor, NoopCursor) else SyncMode.incremental,
+                    catalog_cursor_field,
+                    state,
+                ),
                 max_workers=max_workers,
                 name=stream.name,
+                namespace=stream.namespace,
                 json_schema=stream.get_json_schema(),
                 availability_strategy=StreamAvailabilityStrategy(stream, source),
                 primary_key=pk,
                 cursor_field=cursor_field,
                 slice_logger=source._slice_logger,
                 message_repository=message_repository,
                 logger=logger,
-            )
+                cursor=cursor,
+            ),
+            stream,
+            cursor,
         )
 
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        raise NotImplementedError("This should not be called as part of the Concurrent CDK code. Please report the problem to Airbyte")
```

Comment:
Starting from now, we would like to enforce that state is provided at the initialization and not at any point during the sync. This would simplify instantiation and prevent user from calling `stream.state = X` at the wrong moment